### PR TITLE
fix: use action_mask.sum() for response_length to exclude tool-call tokens in multi-turn training

### LINKED ISSUE #1243: overlong_penalty should exclude tool response tokens from length calculation (use action_mask/action_ranges)
when doing "agentic" training (i.e. training with tools), overlong buffer uses the `experience.response length` as per this following line:

https://github.com/OpenRLHF/OpenRLHF/blob/6c6056daa522e2216466f6e0351cbe453434e185/openrlhf/trainer/ppo_utils/length_penalty.py#L45

response length includes tokens from tool response as per following line:

https://github.com/OpenRLHF/OpenRLHF/blob/3a9815c51aa7e985983eacd06504aaf6db0bbe6b/openrlhf/trainer/ppo_utils/samples_generator.py#L277

given tool responses can't be optimised by model, this can add a bit of noise to "agentic" training and worse can discourage model to use tools.

we should compute response length using `action_mask` or `action_ranges` inside overlong penalty function.

Happy to create PR to fix this. Please let me know if you would like me to.

### PR DESCRIPTION
**Pull request by KamayaniR**
Fixes #1243

## Problem

In `_process_response_into_experience`, `response_length` was computed as the **span** from the first to the last action token:

\`\`\`python
ones_indices = torch.where(action_mask)[0]
response_length = (ones_indices[-1] - ones_indices[0] + 1).item() if len(ones_indices) else 0
\`\`\`

For single-turn training this is fine — the action span is contiguous, so span == sum.

For **multi-turn agentic training** (e.g. `MultiTurnAgentExecutor`), `action_mask` has gaps between turns where tool-call tokens sit. The span overcounts those gaps:

\`\`\`
action_mask: [0 0 0 | 1 1 1 1 | 0 0 0 | 1 1 1 1 1]
                       turn1      tool    turn2
span = 14-3+1 = 12   ← includes 3 tool tokens
sum  = 4+5    = 9    ← only model-generated tokens  ✓
\`\`\`

This causes `apply_overlong_penalty` to measure a longer-than-actual response and trigger false penalties on responses that are not actually overlong.

## Fix

Replace the span with `action_mask.sum()`, which matches the field's documented semantics (`"number of generated tokens"`) and is correct for both single-turn and multi-turn:

\`\`\`python
response_length = action_mask.sum().item()
\`\`\`

## Tests

Added `tests/test_response_length.py` with four cases:
- single-turn (both approaches agree — no regression)
- multi-turn with two turns separated by tool tokens (sum gives 9, span gives 12)
- single-token turns with a gap (sum gives 2, span gives 3)
- empty action mask (both give 0)

### CODE REVIEW

**Comment by KamayaniR** [comment]
Reopening from a dedicated branch to keep this PR isolated from unrelated commits.
